src/Views/Profile/Profile.js

```diff
@@ -1,25 +1,97 @@
 // @flow
-import React, {Component} from 'react';
-import {Paper, RaisedButton} from "material-ui";
-import {PropTypes} from 'prop-types';
+import React, { Component } from 'react';
+import { Paper, RaisedButton, Avatar, FlatButton, TextField } from "material-ui";
+import { PropTypes } from 'prop-types';
+import Messages from "../../Messages";
 
 export default class Profile extends Component {
 
+    constructor(props) {
+        super(props);
+        this.handleEditClick = this.handleEditClick.bind(this);
+        this.handleSaveClick = this.handleSaveClick.bind(this);
+        this.state = { isEditable: false };
+    }
+
     static propTypes = {
         handleLogout: PropTypes.func.isRequired,
-        profile: PropTypes.object.isRequired
+        handleUpdateProfile: PropTypes.func.isRequired,
+        profile: PropTypes.object.isRequired,
     };
 
+    handleEditClick() {
+        this.setState({ isEditable: true });
+    }
+
+    handleSaveClick() {
+        this.setState({ isEditable: false });
+        this.props.handleUpdateProfile(
+            this.refs.textFieldPosition.getValue(),
+            this.refs.textFieldCompanyName.getValue(),
+            this.refs.textFieldPhoneNumber.getValue()
+        );
+    }
+
     render() {
-        const {profile} = this.props;
+        const { profile } = this.props;
+        const isEditable = this.state.isEditable;
+
+        let editButton = null;
+        if (isEditable) {
+            // assign editable profile view
+            editButton = <FlatButton label={Messages.save} style={{ position: 'absolute', right: 0, marginTop: 15 }} onClick={this.handleSaveClick} />;
```

Comment:
to wygląda na to samo, wystarczy zmienic message tak? czyli isEditable ? Messages.edit : Messages.save w label i nie trzeba  ifowc tego buttona, tak? :)